### PR TITLE
Add Dockerfile and Makefile, expand README.md

### PR DESCRIPTION
**Pull request by shaneutt**
This change adds some build tooling for the exporter.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+DOCKER_IMAGE_TAG  ?= latest
+GOOS              ?= $(shell uname -s | tr A-Z a-z)
+GOARCH            ?= $(subst x86_64,amd64,$(patsubst i%86,386,$(shell uname -m)))
+
+all: build
+
+build:
+	@echo ">> building $(GOOS) $(GOARCH)"
+	@go get -v -u ./...
+	@go build -v
```
Comment (mjtrangoni) [inline Makefile]: You could use [promu](https://github.com/prometheus/promu) here.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+DOCKER_IMAGE_NAME ?= postfix-exporter
+DOCKER_IMAGE_TAG  ?= latest
+GOOS              ?= $(shell uname -s | tr A-Z a-z)
+GOARCH            ?= $(subst x86_64,amd64,$(patsubst i%86,386,$(shell uname -m)))
+
+all: build
+
+build:
+	@echo ">> building $(GOOS) $(GOARCH)"
+	@go get -v -u ./...
```
Comment (mjtrangoni) [inline Makefile]: I would use [govendor](https://github.com/kardianos/govendor)

**Comment by shaneutt** [comment]
Are you sure you want *me* to add `govendor` and `promu` to this tool for your team?

I should have mentioned when I made this commit that we use `govendor` and `promu` in our other exporters and projects, but for this PR I was trying to keep things thin, non-intrusive and generally not make any assumptions about how your team would want things done since the repo did not have any version info or contribution documentation.

Let me know your thoughts on the matter, if you don't care I'll add these I just wanted to double check.

**Comment by shaneutt** [comment]
So in our other exporters and projects we use `govendor` and `promu`, but because there was no version info or contribution documentation I wanted to keep this PR lightweight and non-intrusive.

@EdSchouten does the Kumina team want *me* to add all the configurations and documentation for this or would you guys rather do that your own way?

**Comment by timstoop** [comment]
Ed is on honeymoon, so it may take a few weeks for him to respond. And we prefer him to do that, as he wrote the original code! Sorry about that.

That said, if you're willing to add the configurations and documentation, that will surely be appreciated.

Ed has been hesitant regarding adding Makefiles when a simple go build would suffice, to minimise outside dependencies. So you may want to explain why you want this functionality added to the project, in preparation for his response.

**Comment by shaneutt** [comment]
Thanks for the update @timstoop!

> Ed has been hesitant regarding adding Makefiles when a simple go build would suffice, to minimize outside dependencies. So you may want to explain why you want this functionality added to the project, in preparation for his response.

Indeed. This was more about adding back what I added for myself, so if this is unwanted I will not be at all offended if this PR gets closed.

> That said, if you're willing to add the configurations and documentation, that will surely be appreciated.

I'm not opposed to this necessarily, but I feel Ed and/or the organization here may want to do this different than I would ultimately implement it, and have further context on it specific to their production deployment so it might make sense for the owners to decide how they want things done.

> Ed is on honeymoon, so it may take a few weeks for him to respond. And we prefer him to do that, as he wrote the original code! Sorry about that.

No worries, I'll wait to hear from him.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
+FROM golang
```
Comment (EdSchouten) [inline Dockerfile]: Would it make sense to make this Dockerfile more similar to the ones maintained by the Prometheus project, e.g. https://github.com/prometheus/node_exporter/blob/master/Dockerfile ?

**Comment by shaneutt** [comment]
Sounds like this is unwanted, no worries! thanks!

**Pull request closed without merging**